mantidproject/mantid
# Enable use of start and end times as log values

Enable use of `run_start` and `run_end` in muon interfaces.

`run_start` and `run_end` have been added to the list of log values that can be used in the results table in Muon Analysis. They are available either as strings or as numeric values (seconds since start time of first run).

In the ALC interface, using the log values `run_start` and `run_end` now works as it should - the numeric values are seconds since the start of the first run.
- [Online documentation](http://www.mantidproject.org/MuonAnalysisResultsTable#Values) updated for Muon Analysis
- RST documentation updated for ALC
- RST documentation updated for `PlotAsymmetryByLogValue`

**To test:**
1. _Muon Analysis_
   - Open _Interfaces/Muon/Muon Analysis_
   - Load a dataset, for example `MUSR00015189.nxs` from test data (or any muon data)
   - Fit the data on the "Data Analysis" tab. Repeat with a couple of other datasets, using the same fitting model for each.
   - Go to "Results table" tab. In the top table, there should be options to select `run_start` and `run_end` as either text or seconds.
   - Select these options and generate the results table
   - Text should be ISO-formatted date/time string as a label column
   - Seconds should be number of seconds since first run start as an X column
2. _ALC_
   - Open _Interfaces/Muon/ALC_
   - Set first and last datasets, for example `MUSR00015189` and `MUSR00015191` (test data)
   - Set the log value dropdown to either `run_start` or `run_end`
   - Click "Load" and check data is loaded and plotted. The X axis should be number of seconds since the first run start.

Fixes #13195.

[Release notes](https://github.com/mantidproject/mantid/blob/cca8fc32bfbf63955490420561a23676fba5b114/docs/source/release/v3.8.0/muon.rst) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
